airflow/lineage/__init__.py

```diff
@@ -47,20 +47,38 @@ class Metadata:
     data: Dict = attr.ib()
 
 
-def get_backend() -> Optional[LineageBackend]:
-    """Gets the lineage backend if defined in the configs"""
-    clazz = conf.getimport("lineage", "backend", fallback=None)
-
-    if clazz:
-        if not issubclass(clazz, LineageBackend):
-            raise TypeError(
-                f"Your custom Lineage class `{clazz.__name__}` "
-                f"is not a subclass of `{LineageBackend.__name__}`."
-            )
+NoBackend = ...
+
+
+class Backend:
+    _backend = None
+
+    @classmethod
+    def get_backend(cls) -> Optional[LineageBackend]:
+        """Gets the lineage backend lazily if defined in the configs"""
+        if cls._backend == NoBackend:
+            return None
+        elif cls._backend:
+            return cls._backend
+
+        clazz = conf.getimport("lineage", "backend", fallback=None)
+        if clazz:
+            if not issubclass(clazz, LineageBackend):
+                raise TypeError(
+                    f"Your custom Lineage class `{clazz.__name__}` "
+                    f"is not a subclass of `{LineageBackend.__name__}`."
+                )
+            else:
+                cls._backend = clazz()
+                return cls._backend
         else:
-            return clazz()
+            cls._backend = NoBackend
+            return None
+
 
-    return None
+def get_backend():
+    """Gets the lineage backend lazily if defined in the configs"""
+    return Backend.get_backend()
```

Comment:
This change can be made much simpler with `@cache` decorator.

Reply:
Fixed. Thanks!